packages/techdocs-cli/src/commands/helpers.ts

```
@@ -0,0 +1,117 @@
+}
+
+/**
+ * Returns Backstage config suitable for use when instantiating a Publisher. If
+ * there are any missing or invalid options provided, an error is thrown.
+ */
+export const getValidPublisherConfig = (cmd: Command): ConfigReader => {
+  // Assuming that proper credentials are set in Environment variables
+  // for the respective GCS/AWS clients to work.
+  if (
+    !Object.keys(Publisher).includes(
+      cmd.publisherType
+    )
+  ) {
+    throw new Error(`Unknown publisher type ${cmd.publisherType}`);
+  }
+
+  let publisherConfig: PublisherConfig;
+  if (Publisher.azureBlobStorage === cmd.publisherType) {
+    if (!cmd.azureAccountName) {
+      throw new Error(`azureBlobStorage requires --azureAccountName to be specified`);
+    }
+    publisherConfig = {
+      type: cmd.publisherType,
+      [cmd.publisherType]: {
+        containerName: cmd.storageName,
+        credentials: {
+          accountName: cmd.azureAccountName,
+          accountKey: cmd.azureAccountKey
+        }
+      }
+    };
+  } else if (Publisher.awsS3 === cmd.publisherType) {
+    publisherConfig = {
+      type: cmd.publisherType,
+      [cmd.publisherType]: {
+        bucketName: cmd.storageName,
+        ...(cmd.awsRoleArn && { credentials: { roleArn: cmd.awsRoleArn } }),
+        ...(cmd.awsEndpoint && { endpoint: cmd.awsEndpoint }),
+        ...(cmd.awsS3ForcePathStyle && { s3ForcePathStyle: true })
+      }
+    };
+  } else if (Publisher.openStackSwift === cmd.publisherType) {
+    const missingParams = ["osUsername", "osPassword", "osAuthUrl", "osRegion"].filter((param: string) => !cmd[param]);
+
+    if (missingParams.length) {
+      throw new Error(`openStackSwift requires the following params to be specified: ${missingParams.join(', ')}`);
+    }
+
+    publisherConfig = {
+      type: cmd.publisherType,
+      [cmd.publisherType]: {
+        containerName: cmd.storageName,
+        credentials: {
+          username: cmd.osUsername,
```

Should `osUsername`, `osPassword`, `osRegion`, `osAuthVersion`, `osDomainId` and `osDomainName` be added as options to the "migrate" command [here](https://github.com/backstage/techdocs-cli/pull/85/files#diff-d4d6a66778991fda778375618aae1db38a544d1c4b3b3dd2a2ddaff40d55d00fR47)?

---

I think we initially left this off because it's not yet supported?  ...But you're probably right, we should pass these through now and let it fail later (migration not yet implemented).  Less to update when support is added in TechDocs Common!